src/store/Network.ts

```diff
@@ -340,10 +355,29 @@ export default {
                     nodeNetworkModelResult = await networkService
                         .getNetworkModel(currentProfile.selectedNodeUrlToConnect, networkType, isOffline)
                         .toPromise();
+                    let wsUrl = undefined;
+                    let wsConnection = undefined;
+                    if (navigator.onLine && !isOffline) {
+                        wsUrl = nodeNetworkModelResult.repositoryFactory.createListener().url;
+                        wsConnection = await networkService.checkWebsocketConnection(wsUrl);
+                        if (!wsConnection.status && wsConnection.sslNode) {
+                            nodeNetworkModelResult = await networkService
+                                .getNetworkModel(
+                                    currentProfile.selectedNodeUrlToConnect.replace('https', 'http').replace('3001', '3000'),
+                                    networkType,
+                                    isOffline,
+                                )
+                                .toPromise();
+                            wsUrl = nodeNetworkModelResult.repositoryFactory.createListener().url;
+                            wsConnection = await networkService.checkWebsocketConnection(wsUrl);
+                        }
+                    }
```

Comment:
This code block seems to be duplicated. Is it possible to extract this block to an inline arrow function that returns a tuple and use both here(`else` part) and the `if` part?

Reply:
I did refactor this applying the changes in network service.